mediatek: filogic: add support for Totolink X6000R

srt19 opened the pull request:
## Specifications:
-  SOC: MT7981B
- RAM: 256MB
- Flash: 16 MB SPI NOR (EON EN25QX128A)
- Ports: 4 LAN & 1 WAN
- WIFI: MT7981BA + MT7976CN
- LED: 5 WAN (green), status (red/blue)

## Installation via web interface:
Upload the sysupgrade image to firmware upgrade page on OEM web ui.

## Installation via recovery / revert to stock firmware:
Same as [Totolink X5000R](https://openwrt.org/toh/totolink/x5000r#back_to_stock_firmware)

1. Unplug from power
2. Download the Sysupgrade (for openwrt installation) or OEM Factory image (for revert to stock firmware)
3. Set your computer ethernet IP to 192.168.1.X, subnet 255.255.255.0 (X=your chosen ip number ranging from 2-254)
4. Press and hold reset button while turn on / plug power adapter to the router. Wait untill ALL LAN green LED turn on, then release the reset button.
5. Plug the LAN cable from your computer into LAN 1 port  then open web browser, and type in the address column : 192.168.1.1
6. Upload the sysupgrade / oem factory image
7. Wait until the router finished flashing (the router will reboot)
8. Set ethernet ip to dhcp
9. Open 192.168.1.1 (for openwrt install) / 192.168.0.1 (for OEM firmware)

srt19 commented:
> Can you share the output of `hexdump -Cs 0x19a -n 1 /dev/mtd2`
Here's the output.
<img width="1482" height="762" alt="image" src="https://github.com/user-attachments/assets/4c3aadbf-5761-46d2-a1f1-6060c7585f86" />

castiel652 commented:
Cool thanks. :)

castiel652 commented on target/linux/mediatek/dts/mt7981b-totolink-x6000r.dts:
```diff
@@ -0,0 +1,249 @@
+};
+
+&wifi {
+	status = "okay";
```
Where's the Wi-Fi mac address?

srt19 replied:
I believe it's using the eeprom.
I test both using the mac address cell + eeprom and eeprom only and it gives the same mac address

castiel652 replied:
I see.

castiel652 commented:
You need to squash those two commits into one and fix the commit message.

srt19 commented:
> You need to squash those two commits into one and fix the commit message.

Umm...
How to squash the commit ?
I tried to rebase it and squash it, is it the correct way ?

castiel652 commented:
`git rebase -i HASH`
HASH is the commit hash before X6000R commit.

srt19 commented:
> `git rebase -i HASH` HASH is the commit hash before X6000R commit.

Am I doing it right?

castiel652 commented:
No. Did you read the screen after issuing the command

srt19 commented:
@castiel652 Yes, I do select squash all of the commit except the first one

castiel652 commented:
But there's now several commits...
And why do you keep merging main back to your branch? You should rebase on top of it instead.

srt19 committed:
mediatek: filogic: add support for Totolink X6000R

Specification :
- SOC : Mediatek MT7981B (1.3GHz)
- RAM : 256MB
- Flash : 16MB SPI NOR
- Ports : 4 LAN (1G) & 1 WAN (1G)
- WIFI : MediaTek dual-band WiFi 6
  - 2.4 GHz : b/g/n/ax, MIMO 2x2
  - 5 GHz   : a/n/ac/ax, MIMO 2x2
- Buttons : Reset & WPS/Mesh
- LEDS : WAN (Green), Status (Red & Blue)
- Power : 12V1A

Install via OEM web ui:
Upload the sysupgrade image to firmware upgrade page on OEM web ui

Install via recovery / revert to stock firmware:
1. Unplug from power
2. Download the Sysupgrade (for openwrt installation) or OEM Factory image (for revert to stock firmware)
3. Set your computer ethernet IP to 192.168.1.X, subnet 255.255.255.0 (X=your chosen ip number ranging from 2-254)
4. Press and hold reset button while turn on / plug power adapter to the router. Wait untill ALL LAN green LED turn on, then release the reset button.
5. Plug the LAN cable from your computer into LAN 1 port then open web browser, and type in the address column : 192.168.1.1
6. Upload the sysupgrade / oem factory image
7. Wait until the router finished flashing (the router will reboot)
8. Set ethernet ip to dhcp
9. Open 192.168.1.1 (for openwrt install) / 192.168.0.1 (for OEM firmware)

Signed-off-by: Ari Kurniawan <noobhek@gmail.com>
Link: https://github.com/openwrt/openwrt/pull/20035
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

castiel652 commented on target/linux/mediatek/dts/mt7981b-totolink-x6000r.dts:
```diff
@@ -0,0 +1,249 @@
+	chosen {
+		stdout-path = "serial0:115200n8";
+	};
+
```
```suggestion
        memory@40000000 {
		reg = <0 0x40000000 0 0x10000000>;
		device_type = "memory";
	};
	
```

castiel652 reviewed (changes requested):
And fix the commit message.
